daemon/network.go

```diff
@@ -347,9 +347,14 @@ func (daemon *Daemon) createNetwork(cfg *config.Config, create types.NetworkCrea
 		nwOptions = append(nwOptions, libnetwork.NetworkOptionConfigOnly())
 	}
 
-	if err := network.ValidateIPAM(create.IPAM, create.EnableIPv6); err != nil {
+	if network.HasIPv6Subnets(create.IPAM) && !create.EnableIPv6 {
+		create.EnableIPv6 = true
+	}
```

Comment:
Overall this looks fine; starting to wonder now if we should have something that _returns_ the IPAM config;

- `ValidateIPAM` also loops over all configs (including checking for IPv6)
- But the only thing returned is an error (i.e., we throw away some of the work)

We could have some function that handles the IPAM config, returns it (which may include setting some additional "runtime" data perhaps)

```go
ipamConfig, err := foo.ParseIPAMConfig()
```

Or perhaps this should be a `libnetwork.NetworkOptionIPAM(create.IPAM)`. We should update `NetworkOption` to return an error (maybe I already had something like that in one of my PRs), so that it can return an error if the option was invalid.

Eitherway; that's definitely fine to be looked at separately 😅